### PR TITLE
Add separate rule for Kubernetes capitalization

### PR DESCRIPTION
**Pull request by jdbaldry**

### CODE REVIEW

**Comment by clayton-cornell** [comment]
Is this the right place to define more K8s-specific terminology?  For example:
* Kubectl
* Kubelet
* Minikube
* etc.

**Comment by jdbaldry** [comment]
Good question. This is a substitution rule so we can definitely add some substitutions here :)

- Kubectl: I think we should suggest only "`kubectl`" (https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-kubernetes-command-tool-and-component-names)
- Kubelet: similarly, only "`kubelet`" (https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-kubernetes-command-tool-and-component-names)
- Minikube: it looks like the preferred name in the project is "minikube" (https://minikube.sigs.k8s.io/docs/) which would need to also be added to the dictionary so the spellchecker knows of it.

**Comment by clayton-cornell** [comment]
Oooops, they got capitalized in my list :-)  I'm not so sure we use `minikube` in our docs, but within Agent, we use `kubectl` and `kubelet` a lot.

**Comment by clayton-cornell** [comment]
Will these fail if they are lowercase in a component name?  Example: `k8s.daemonset.name` is an attribute in the `otelcol.processor.k8sattributes` component.

**Comment by jdbaldry** [comment]
Not in `inline code`, or **bold**, we've configured Vale to ignore those tokens because of https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#bold

File: vale/fixtures/Grafana/Kubernetes/testinvalid.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Don't use:
+
+- cron job
+- deamonset
+- Kubernetes deployment
+- pod
+- replicaset
+- statefulset
+- kubelet
+- kubectl
```
Comment (clayton-cornell) [inline vale/fixtures/Grafana/Kubernetes/testinvalid.md:10]: Do we need tests for Kubectl and Kubelet (with initial caps)?

Comment (jdbaldry) [reply vale/fixtures/Grafana/Kubernetes/testinvalid.md:10]: Good catch, added in [2f9a383](https://github.com/grafana/writers-toolkit/pull/512/commits/2f9a3839edf1bc75362eb3e03774ad1d1293e0d2)

**Review by clayton-cornell — APPROVED** [review]
This looks great to me now :-)

**Pull request merged**